frontend/src/components/table/Table.jsx

```diff
@@ -7,27 +7,96 @@ import {
-            <ScrollTableButton table-id={tableId} />
-            <OverflowWrapper id={tableId}>
-                <SpreadSheetTable>{props.children}</SpreadSheetTable>
-            </OverflowWrapper>
-        </TableWrapper>
-    );
-};
+    const [canScrollLeft, setCanScrollLeft] = useState(false);
+    const [canScrollRight, setCanScrollRight] = useState(false);
+    const [scrollPosition, setScrollPosition] = useState(0);
+    const [windowSize, setWindowSize] = useState([0, 0]);
+    const maxScrollLeft = React.useRef(0);
 
-export const TableSimple = props => {
-    const tableId = props['table-id'];
+    // Checks if table is scrolled using other methods than scrollButtons (e.g. keyboard / mouse) ==> scrollPosition
+    useEffect(() => {
+        const savedRef = ref.current;
+
+        function updateScrollPosition() {
+            setScrollPosition(savedRef.scrollLeft);
+        }
+
+        savedRef.addEventListener('scroll', updateScrollPosition);
+        updateScrollPosition();
+
+        return () =>
+            savedRef.removeEventListener('scroll', updateScrollPosition);
+    }, [ref]);
+
+    // Checks if window is resized ==> windowSize
+    useEffect(() => {
+        function updateWindowSize() {
+            setWindowSize([window.innerWidth, window.innerHeight]);
+        }
+
+        window.addEventListener('resize', updateWindowSize);
+        updateWindowSize();
+
+        return () => window.removeEventListener('resize', updateWindowSize);
+    }, []);
+
+    // Checks if table is scrollable and decides to which directions it's scrollable
+    useEffect(() => {
+        maxScrollLeft.current =
+            ref.current.scrollWidth - ref.current.clientWidth;
+
+        // check that table is scrollable and it has not been scrolled
+        if (maxScrollLeft.current !== 0 && scrollPosition === 0) {
+            setCanScrollRight(true);
+            setCanScrollLeft(false);
+        }
+
+        // show scroll left and right button
+        if (scrollPosition !== 0) {
+            setCanScrollLeft(true);
+            setCanScrollRight(true);
+        }
+
+        // hide scroll right button when there's nothing to scroll
+        if (
+            scrollPosition === maxScrollLeft.current ||
+            scrollPosition === maxScrollLeft.current - 1
+        ) {
+            setCanScrollRight(false);
+        }
+    }, [ref, scrollPosition, windowSize]);
+
+    function moveRight() {
+        ref.current.scrollLeft += 150;
+        setScrollPosition(scrollPosition + 150);
+    }
+
+    function moveLeft() {
+        ref.current.scrollLeft -= 150;
+        setScrollPosition(scrollPosition - 150);
+    }
 
     return (
         <TableWrapper>
-            <ScrollTableButton table-id={tableId} />
-            <OverflowWrapper id={tableId}>
-                <SimpleTable>{props.children}</SimpleTable>
+            <ScrollTableButton
+                can-scroll-left={canScrollLeft}
+                can-scroll-right={canScrollRight}
```

Comment:
Reason for that these props are written different ways? `can-scroll-left` vs `moveRight`? To say other is boolean and other is function?

Reply:
No, this was only a mishap 😄